### PR TITLE
Move the removal/unlinking of the deposit directory content.

### PR DESCRIPTION
**Pull request by justinlittman**
refs #310

## Why was this change made?
Hope that it will alleviate pres robots and pres cat issues with reading from Ceph backed storage.

## How was this change tested?
Unit

## Which documentation and/or configurations were updated?
NA

### CODE REVIEW

File: spec/preservation_ingest/complete_ingest_spec.rb [hunk 1]
```diff
@@ -42,7 +15,6 @@
     end
 
     it 'removes the deposit bag and updates the accessionWF when no errors are raised' do
```
Comment (jmartin-sul) [inline spec/preservation_ingest/complete_ingest_spec.rb:17]: we should strike the "removes the deposit bag" part from the title now

File: spec/preservation_ingest/update_catalog_spec.rb [hunk 2]
```diff
@@ -18,10 +20,15 @@
     }
   end
   let(:mock_storage_object) do
-    instance_double(Moab::StorageObject, object_pathname: instance_double(Pathname), size: size, storage_root: strg_root, current_version_id: version)
+    instance_double(Moab::StorageObject, deposit_bag_pathname: deposit_bag_pathname, object_pathname: mock_moab_pathname, size: size, storage_root: strg_root, current_version_id: version)
   end
   let(:mock_storage_objects) { [mock_storage_object] }
 
+  let(:spec_dir) { File.join(File.dirname(__FILE__), '..') }
+  let(:deposit_dir_pathname) { Pathname(File.join(spec_dir, 'fixtures', 'deposit', 'complete-ingest')) }
```
Comment (jmartin-sul) [inline spec/preservation_ingest/update_catalog_spec.rb:28]: i'd consider changing `'complete-ingest'` to `'update-catalog'` or `'deposit_cleanup'`, but it also only lives for the duration of the test, so not a big deal either way

**Review by jmartin-sul — APPROVED** [review]
👍 pending consideration of the couple nitpicks i left (and testing outcome of course).

File: spec/preservation_ingest/complete_ingest_spec.rb (hunk 1 above)
Comment (justinlittman) [reply spec/preservation_ingest/complete_ingest_spec.rb:17]: Struck.

File: spec/preservation_ingest/update_catalog_spec.rb (hunk 2 above)
Comment (justinlittman) [reply spec/preservation_ingest/update_catalog_spec.rb:28]: Renamed.

**Review by jmartin-sul — APPROVED** [review]

**Comment by jmartin-sul** [comment]
This has been deployed to stage for most of the last three weeks (I first deployed it to stage May 6, and rebased on `main` and re-deployed to stage each week after dependency update testing for the week finished):  https://app.honeybadger.io/projects/55564/deploys?environment=stage

I queried for the druids of every Moab that was created or updated on stage since that first deployment:
```ruby
may_six = Date.parse('2021-05-06')
druids_updated_since_may_6 =
  CompleteMoab.
  joins(:preserved_object).
  where(created_at: (may_six..Time.now)).or(
    CompleteMoab.where(updated_at: (may_six..Time.now))).
  pluck(:druid)
```

I ran checksum validation over all of them; the audit completed successfully for each, and all came back `ok`.

I also ran slightly modified versions of the queries to check for replication issues (see [db/README.md](https://github.com/sul-dlss/preservation_catalog/tree/4be2e120233dec235b9afb5fb230156bc6b08b4f/db#which-druids-dont-have-a-zipped-moab-version-for-each-of-their-versions-on-each-of-their-target-endpoints)), limited to the modified druid list, and turned up no results:
```ruby
# which druids don't have a zipped moab version for each of their versions on each of their target endpoints?

## which have some of the expected versions, but not all?
PreservedObject.
  joins(preservation_policy: [:zip_endpoints]).
  left_outer_joins(:zipped_moab_versions).
  where('zip_endpoints.id = zipped_moab_versions.zip_endpoint_id').
  where(druid: druids_updated_since_may_6).
  group(:druid, :current_version, :endpoint_name).
  having('count(zipped_moab_versions.id) < current_version').
  pluck(:druid, :current_version, :endpoint_name, 'count(zipped_moab_versions.id)')

## which have none of the expected versions?
PreservedObject.
  joins(preservation_policy: [:zip_endpoints]).
  left_outer_joins(:zipped_moab_versions).
  where(druid: druids_updated_since_may_6).
  group(:druid, :current_version, :endpoint_name).
  having('count(zipped_moab_versions.id) = 0').
  pluck(:druid, :current_version, :endpoint_name, 'count(zipped_moab_versions.id)')

# look for non-ok zip parts, and other replication irregularities for druids that have started replication
druids_with_replication_issues =
  ZipPart.
    joins(zipped_moab_version: [{ preserved_object: [:complete_moabs] }, :zip_endpoint]).
    where(preserved_objects: { druid: druids_updated_since_may_6 }).
    where.not(zip_parts: { status: 'ok' }).
    group(:druid, :endpoint_name).
    pluck(:druid,
          Arel.sql('SUM(zip_parts.size) AS zmv_size'),
          Arel.sql('max(complete_moabs.size) AS moab_size'), # max(complete_moabs.size) -- heuristic is that biggest moab is usually latest version is usually most up to date, since things mostly work
          Arel.sql('SUM(zip_parts.size) - max(complete_moabs.size) AS zmv_size_diff'),
          :endpoint_name,
          Arel.sql('COUNT(zip_parts.id) parts_count'),
          Arel.sql('ARRAY_AGG(zip_parts.status) part_statuses'))
```

I also did the update-many-times-in-quick-succession test Andrew describes here, and was able to see the test druid being versioned and preserved properly.  It also passed a manual run of checksum validation.  see https://argo-stage.stanford.edu/view/druid:bb557jq7776

We did see a few occurrences of a version mismatch error on the `sdr-ingest-transfer` step, but we've seen this issue occassionaly prior to the change in this PR, the rate isn't any higher the last few weeks, and we doubt this change would play a part in that, so that doesn't seem like a blocker to deploying this.

Discussed with @andrewjbtw this afternoon, and he's going to do one last test where he accessions a large (> 100 GB) object.  If that doesn't turn up any worrisome issues (we don't expect it to 🤞), we'll merge this and deploy it to prod next week, after the dependency updates go out (in the morning, with at least a couple hours between this and dep updates, just to help disentangle any possible prod issues we might see).

**Pull request merged**